src/with-layout.js

```diff
@@ -0,0 +1,22 @@
+const layoutSymbol = Symbol('layout');
+
+const withLayout = (mapToLayout) => (Component) => {
+    const value = typeof layout === 'function' ? mapToLayout : () => mapToLayout;
```

Comment:
```suggestion
    const value = typeof mapToLayout === 'function' ? mapToLayout : () => mapToLayout;
```
Right?

Reply:
Nice catch! This was a bug. I've fixed it and add more tests.